contrib/mesos/pkg/executor/executor.go

```diff
@@ -1044,3 +1002,220 @@ func executorInfoToAnnotations(ei *mesos.ExecutorInfo) (annotations map[string]s
 
 	return
 }
+
+// implements txTaskOperator, uses the executor state lock to handle concurrent invocation
+func (k *Executor) doForTask(taskId string, f func(*kuberTask, *api.Pod) bool) (changed bool, err error) {
+	k.lock.Lock()
+	defer k.lock.Unlock()
+
+	// exclude tasks which are already deleted from our task registry
+	task := k.tasks[taskId]
```

Comment:
Can you abstract this into a TaskRegistry which has an modify(taskId, modifier) or update(taskId, updater) method? This transaction logic is nicely orthogonal to everything that the executor.go actually does.